clients/src/main/java/org/apache/kafka/clients/ClientTelemetryUtils.java

```diff
@@ -358,4 +360,15 @@ public static int calculateQueueBytes(ApiVersions apiVersions,
         }
     }
 
+    public static MetricsData deserializeMetricsData(ByteBuffer serializedMetricsData) {
+        MetricsData metricsData = null;
+
+        try {
+            ByteBuffer metricsBuffer = (ByteBuffer) serializedMetricsData.flip();
+            metricsData = MetricsData.parseFrom(metricsBuffer);
+        } catch (IOException e) {
+            log.warn("Unable to parse MetricsData payload: {} ", e.getMessage());
```

Comment:
For this one I didn't throw an exception because I use this method on the [exportMetrics()](https://github.com/confluentinc/kafka/pull/779/files#diff-0715381a0a75ad5cd2c7184c08b802cfeb783904f680dfbdad6f39a2e2d0a62eR118) method that the broker calls and there I just return or skip the export if I dont have valid metrics to send.